File: sorting-algorithms/C/Insertion Sort/InsertionSort.c
#include <stdio.h>

void insertionSort(int element[],int size){
    for(int step = 1; step < size; step++){
        int key = element[step];
        int j = step - 1;

        // Compare key with each element on the left of it until an element smaller than it is found.
        while(key < element[j] &&j <= 0){        
        }
    };  
};

int main(){
    int element[] = {9, 5, 1, 4, 3};
    int size = sizeof(element) / sizeof(element[0]);

    // Original Array Before Performing Insertion.
    printf("Original Array: \n");
    for(int i = 0; i < size;++i){
        printf("%d",&element[i]);
    };
    printf("\n"); 

    // Perform Insertion Sort.
    insertionSort(element,size);

    printf("Sorted Array In An Ascending Order:\n");
    for(int i =0; i < size; ++i){
        printf("%d",&element[i]);
    };
};

File: python/sorting_algorithms/Insertion Sort/insertionSort.py
def insertionSort(array):
    for step in range(1,len(array)):
        key = array[step]
        j = step - 1
        # Compare key with each element on the left of it until an element smaller than it is found
        while j >= 0 and key < array[j]:
            array[j + 1] = array[j]               
            j -= 1
        # Place key at after the element just smaller than it. 
        array[j + 1] = key 

array = [19, 5, 13, -4, 3,10,0,11,2]
print(array)

# Peform Insertion Sort for each element in the array
insertionSort(array)
print("Sorted Array In Ascending Order:\n",array)